### PR TITLE
Adds functionality to gather translation strings from yoastseo via webpack

### PR DESCRIPTION
**Pull request by mhkuu**
## Context

* Now that Premium- and product-specific assessments are properly bundled, we should also update their translation domains. With this PR, we add functionality for Yoast SEO for WooCommerce to fetch these translation strings from the `yoastseo` package in `wordpress-seo` through `webpack`.

## Summary

This PR can be summarized in the following changelog entry:

* Adds functionality to gather translation strings from yoastseo via webpack.

## Relevant technical choices:

* The implementation mimics that of `wordpress-seo-premium`, so for code review, it's best to compare the two implementations and see if nothing is missing. 

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Require the Free branch using `composer require yoast/wordpress-seo:dev-fix-translation-bundling@dev --dev`.
* Confirm that, in this branch, `languages/yoast-woo-seo.pot` now additionally contains the translation strings for e-commerce assessments (_list_, _image alt tags_, _product identifier_, and _SKU_ assessments).
* Run `composer install && yarn && grunt build` to find that the translations file will be updated with only the creation date, confirming that these translation strings are now properly added. 

### Test instructions for QA when the code is in the RC

* [ ] QA should use the same steps as above.

QA can test this PR by following these steps:

* QA can skip checking that the translations are updated.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* Generation of the `languages/yoast-woo-seo.pot` file.

## UI changes

* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Documentation

* [ ] I have written documentation for this change.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

## Innovation

* [x] No innovation project is applicable for this PR.
* [ ] This PR falls under an innovation project. I have attached the `innovation` label and noted the work hours.

Fixes https://github.com/Yoast/plugins-automated-testing/issues/648

### CODE REVIEW

**Review by d-claassen — COMMENTED** [review]
Code review done. Changes so far look pretty much as expected, just 2 concerns:

1. I expect running `$ grunt build:js` directly will now create a new `gettext.pot` file in the root of the repo. In Free, that file is being ignored from Git. See [this line in that `.gitignore` file](https://github.com/Yoast/wordpress-seo/blob/trunk/.gitignore#L79).
1. The actual introduction of the new translations from the assessments is not in this PR. I wonder if that should be here?

----

There are some Grunt configuration changes. They seem sensible, but the usual Grunt commands should be tested with care. I don't know the exact expectations for them, but the usual devs touching this should know.

- `grunt build`
- `grunt build:js`
- `grunt build:i18n` _optional. Probably not used on its own very often._

**Comment by mhkuu** [comment]
> * I expect running `$ grunt build:js` directly will now create a new `gettext.pot` file in the root of the repo. In Free, that file is being ignored from Git. See [this line in that `.gitignore` file](https://github.com/Yoast/wordpress-seo/blob/trunk/.gitignore#L79).

Actually, no, this file is created but its contents are pasted into the `languages/yoast-woo-seo.pot` file 😄 

> * The actual introduction of the new translations from the assessments is not in this PR. I wonder if that should be here?

Good point, I'll add that here as well as in `wordpress-seo-premium`.

File: package.json [hunk 1]
```diff
@@ -59,21 +59,23 @@
...
     "jest": "^27.2.2",
     "lodash.defaultsdeep": "^4.6.0",
     "node-fetch": "^1.0.1",
     "unminified-webpack-plugin": "^2.0.0",
-    "webpack": "^4.16.2",
+    "webpack": "^4.20.2",
     "webpack-bundle-analyzer": "^4.8.0",
-    "webpack-cli": "^3.1.0",
+    "webpack-cli": "^3.1.2",
     "yoastseo": "link:vendor/yoast/wordpress-seo/packages/yoastseo"
```
[the remark this answers is not in the document]
Comment (mhkuu) [reply package.json:79]: Yeah -- but I'm not going to touch that here 😅 Maybe something to discuss with the technical council.

**Comment by agnieszkaszuba** [comment]
CR: Everything looks good, except that I noticed that the strings 'Barcode' and 'Product identifier' are not in the .pot file, despite the fact that the domain for those strings was changed in the [Free PR](https://github.com/Yoast/wordpress-seo/blob/6d673c2d1692c6c4025383de16dc4174f0ce3ee8/packages/yoastseo/src/scoring/assessments/seo/ProductIdentifiersAssessment.js#L34).

However, this might be an existing issue. I don't see those strings on translate.wordpress.org either. Maybe because the string is created from a variable, creating the translation string doesn't work?

**Comment by mhkuu** [comment]
Thanks, that's pre-existing indeed :-( Using a variable in a `__` string is not going to work, unfortunately. Let's resolve that in another task.

**Pull request merged**